staging/src/k8s.io/apiserver/pkg/server/options/admission.go

```diff
@@ -94,3 +103,29 @@ func (a *AdmissionOptions) Validate() []error {
 	errs := []error{}
 	return errs
 }
+
+// preparePluginNamesIfNotProvided sets admission plugin names if they
+// were not provided by a cluster-admin. This method makes use of RecommendedPluginOrder
+// as well as DefaultOffPlugins fields.
+func (a *AdmissionOptions) preparePluginNamesIfNotProvided() {
+	if len(a.PluginNames) > 0 {
+		return
+	}
+	a.PluginNames = a.RecommendedPluginOrder
+
+	onlyEnabledPluginNames := []string{}
+	for _, pluginName := range a.PluginNames {
+		disablePlugin := false
+		for _, disabledPluginName := range a.DefaultOffPlugins {
+			if pluginName == disabledPluginName {
+				disablePlugin = true
+				break
+			}
+		}
+		if !disablePlugin {
```

Comment:
nit: preference for `if disablePlugins { continue; }` so I don't have to parse "not disabled plugin" in my head.